### DISCUSSION TITLE
Using ufo3 file format to work across different font editing software

### DISCUSSION THREAD

**Discussion by jacobxperez** [jacobxperez/lexica-ultralegible#11] [Ideas]
Will it be more convenient to use [ufo3](https://unifiedfontobject.org/versions/ufo3/) file format to work across different font editing software, it may be a better option, this can also increase the contribution to this project by including people who use different software.
I found this, the [Font Development Best Practices](https://silnrsi.github.io/FDBP/en-US/UFO.html) on GitHub.

**Comment by jacobxperez** [comment]
@RandomMaerks I started this discussion, let me know what you think.

**Comment by RandomMaerks** [comment]
Generally, I like the idea of including .ufo files. As you've mentioned, it does work for multiple font editing software, and can also be converted to other formats.

However, I think we should keep the .sfd files as an alternate format for FontForge users. FF sometimes doesn't read .ufo files properly, and certain features might be lost.

**Comment by RandomMaerks** [comment]
Also, based on my personal experience, uploading .ufos to GitHub is a bit of an inconvenience.
GitHub limits the number of files uploaded at once, and one .ufo folder usually contains about the amount of glyphs in the font plus some additional files.
There are third-party programs and various other methods to upload in bulk, but it might require some time and understanding to set up, especially for people who aren't super tech-savvy.

Yes, I'm kinda bullcrapping about an issue that noone even knows about, but that's just my opinion

**Reply by jacobxperez** [reply]
Alright then, lets just leave it as it is for now, this is something that requires further research. I will keep you updated.

**Comment by RandomMaerks** [comment]
7 months have passed. I have realised that there is a web-based editor on GitHub, enabled by pressing the full-stop key while opening the repo or changing the web extension from .com to .dev. There is no limit in the number of files uploaded, and the total file size must be under 2GB each push.

Uploading .ufos this way is a lot easier, and each .ufo should be around 1-5 megabytes depending on how big the font is. The only issue is upload time, but it usually isn't that long if you have a stable connection.

I think we should start including .ufos in the repo using the web editor.

**Reply by jacobxperez** [reply]
@RandomMaerks Sorry for not responding earlier I was sick. This sounds great are there any other font Project implementing this idea? Also is there any tutorial I don't want to implement something without knowing?
